diff --color -urN vcmi-1.5.7/scripts/lib/erm/MF.lua vcmi/scripts/lib/erm/MF.lua
--- vcmi-1.5.7/scripts/lib/erm/MF.lua	2024-08-25 23:27:47.000000000 +0200
+++ vcmi/scripts/lib/erm/MF.lua	2024-12-19 15:00:22.988151657 +0100
@@ -9,7 +9,7 @@
 end
 
 function MF:D(x)
-	return self.ERM.activeEvent:getInitalDamage()
+	return self.ERM.activeEvent:getInitialDamage()
 end
 
 function MF:E(x, ...)
diff --color -urN vcmi-1.5.7/scripts/lib/erm/UN.lua vcmi/scripts/lib/erm/UN.lua
--- vcmi-1.5.7/scripts/lib/erm/UN.lua	2024-08-25 23:27:47.000000000 +0200
+++ vcmi/scripts/lib/erm/UN.lua	2024-12-19 15:00:22.988151657 +0100
@@ -247,7 +247,7 @@
 	error ("UN:T is not implemented")
 end
 
--- count objects, get coordiantes
+-- count objects, get coordinates
 function UN:U(x, ...)
 	error ("UN:U is not implemented")
 end
